tests/trilinos/vector_reference_01.cc

```diff
@@ -0,0 +1,95 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+
+// TrilinosWrappers::internal::VectorReference had its non-const copy
+// operator return a const reference. that was non-intuitive but,
+// because of the particular semantics of copying vector reference
+// objects, made no difference. either way, have a check for these
+// semantics
+
+#include "../tests.h"
+#include <deal.II/base/utilities.h>
+#include <deal.II/lac/trilinos_vector.h>
+#include <fstream>
+#include <iostream>
+
+
+void test ()
+{
+  TrilinosWrappers::Vector v(3);
+  v(0) = 0;
+  v(1) = 1;
+  v(2) = 2;
+
+  TrilinosWrappers::internal::VectorReference a (v(0));
+  TrilinosWrappers::internal::VectorReference b (v(1));
+  TrilinosWrappers::internal::VectorReference c (v(2));
+
+  // copy VectorReference objects. note that operator= for these
+  // objects does not copy the*content* of the reference object, but
+  // assigns the value of the vector element referenced on the right
+  // to the vector element referenced on the left
+  (c = a) = b;
+  b = a;
+
+  deallog << static_cast<TrilinosScalar>(a) << std::endl; // should point to v(0)
+  deallog << static_cast<TrilinosScalar>(b) << std::endl; // should point to v(0)
+  deallog << static_cast<TrilinosScalar>(c) << std::endl; // should point to v(0)
```

Comment:
This comment does not match the output: we print `1.000`, not `0`, for `c`.

I think the the comment is wrong and the output is right. Lines 46 and 47 should do the following:
1. assign `c` to point to where `a` points (`c` points to entry 0)
2. assign `c` to point to where `b` points (`c` points to entry 1)
3. assign `b` to point where `a` points (`b` points to entry 0)

Is this correct?